Add section on cert caching (#19195)
dotnet/AspNetCore.Docs: aspnetcore/security/authentication/certauth.md

{
    services.AddAuthentication(
        CertificateAuthenticationDefaults.AuthenticationScheme)
        .AddCertificate()

> 	// Adding a ICertificateValidationCache will result in certificate auth caching the results, the default implementation uses a memory cache

Reviewer: Adding `AddCertificateCache` to this sample makes it specific to 5.0

{
    services.AddAuthentication(
        CertificateAuthenticationDefaults.AuthenticationScheme)
::: moniker range="< aspnetcore-5.0"
        .AddCertificate();
::: moniker-end
::: moniker range=">= aspnetcore-5.0"
        .AddCertificate()
        // Adding a ICertificateValidationCache will result in certificate auth caching the results, the default implementation uses a memory cache

>         .AddCertificateCache()

Reviewer: ```suggestion
        .AddCertificateCache();
```

<a name="occ"></a>

> ## Certificate validation cachine

Reviewer: There is probably conditional display instruction to only display this content when viewing 5.0 docs, e.g.

`::: moniker range=">= aspnetcore-3.0"`

`::: moniker-end`

Should use this so we don't need to mention the version it is available from. I don't know the variable is called in 5.0

Reviewer 2: As James mentioned, you can toggle the visibility of content for 5.0 and later with the following:

```
::: moniker range=">= aspnetcore-5.0"

5.0+ content here

::: moniker-end
```

> ## Certificate validation cachine

Reviewer 2: ```suggestion
## Certificate validation caching
```

> ASP.NET Core 5 preview 7 and later adds the ability to enable caching of validation results which drastically improves performance as validation is a expensive operation.

Reviewer 2: ```suggestion
ASP.NET Core 5.0 Preview 7 and later adds the ability to enable caching of validation results. The caching dramatically improves performance, as validation is an expensive operation.
```

To enable the the default implementation that uses Memory cache, just call:

`.AddCertificate().AddCertificateCache(o => { o.CacheSize = 1024; o.CacheEntryExpiration = TimeSpan.FromMinutes(2); })`

You also can provide in your own implementation via `services.AddSingleton<ICertificateValidationCache, YourCache>()`

> 
> 

Reviewer 2: Remove the blank lines.

```suggestion
```

<a name="occ"></a>

::: moniker range=">= aspnetcore-5.0"
## Certificate validation caching

> ASP.NET Core 5.0 Preview 7 and later adds the ability to enable caching of validation results. The caching dramatically improves performance, as validation is an expensive operation.

Reviewer: ```suggestion
ASP.NET Core 5.0 Preview 7 and later adds the ability to enable caching of validation results. The caching dramatically improves performance of certificate authentication, as validation is an expensive operation.
```
